File: 20. Valid Parentheses/20. Valid Parentheses.cpp
class Solution {
public:
    bool isValid(string s) {
        stack<char> parentheses;
        for(auto iter = s.begin(); iter != s.end(); ++iter){
            if(*iter == '(' || *iter == '[' || *iter == '{')	parentheses.push(*iter);
            else if(parentheses.empty())	return false;
            else{
                char &tmp = parentheses.top();
                if((tmp == '(' && *iter == ')') || (tmp == '[' && *iter == ']') || (tmp == '{' && *iter == '}'))
                    parentheses.pop();
                else	return false;
            }
        }
        return parentheses.empty();
    }
};

File: 20. Valid Parentheses/20. Valid Parentheses.java
class Solution {
    public boolean isValid(String s) {
        Stack<Character> st = new Stack<>();
        for(Character c: s.toCharArray()){
            if(c == '{' || c == '[' || c == '('){
                st.push(c);
            }
            else if( st.empty() || (c == '}' && st.peek() != '{') ||
                (c == ']' && st.peek() != '[') ||
                (c == ')' && st.peek() != '(')){
                return false;
            }
            else{
                st.pop();
            }
        }
        return st.empty();
    }
}